src/lvm.cpp

```diff
@@ -0,0 +1,451 @@
+        throw std::runtime_error(
+            "LVM ERROR: Failed to list available partitions.");
+    }
+}
+void LVM::checkLVMAvailability()
+{
+    LOG_INFO("LVM: Begin of availability check.")
+    verifyBootIsNotLVM();
+    checkUEFIMode();
+    checkLVMEnabled();
+    checkThinProvisioning();
+    checkEnoughDiskSpaceAvailable();
+    verifyAvailablePartitions();
+    LOG_INFO("LVM: End of availability check.")
+}
+
+bool LVM::snapshotExists(const std::string& snapshotName)
+{
+    std::list<std::string> output;
+
+    const std::string checkSnapshotCommand
+        = fmt::format("lvs --noheadings -o lv_name {}/{}",
+            m_snapshotVolumeGroup, snapshotName);
+    int exitCode = cloyster::runCommand(checkSnapshotCommand, output, false);
+
+    // If the command succeeded (exitCode 0) and output is not empty, snapshot
+    // exists
+    if (exitCode == 0 && !output.empty()) {
+        std::string lvName = output.front();
+
+        // Trim leading/trailing whitespaces
+        lvName.erase(lvName.begin(),
+            std::find_if(lvName.begin(), lvName.end(),
+                [](unsigned char ch) { return !std::isspace(ch); }));
+        lvName.erase(std::find_if(lvName.rbegin(), lvName.rend(),
+                         [](unsigned char ch) { return !std::isspace(ch); })
+                         .base(),
+            lvName.end());
+
+        // Check if the name matches the expected snapshot name
+        if (lvName == snapshotName) {
+            LOG_INFO("LVM: Snapshot {} exists.", snapshotName);
+            return true;
+        } else {
+            LOG_INFO("LVM: Snapshot found but name mismatch: expected '{}', "
+                     "got '{}'",
+                snapshotName, lvName);
+        }
+    }
+
+    // If the command failed or the snapshot was not found
+    LOG_INFO("LVM: Snapshot {} does not exist.", snapshotName);
+    return false;
+}
+
+void LVM::verifyBootIsNotLVM()
+{
+    std::list<std::string> output;
+
+    const std::string listCommand
+        = "lsblk --noheadings --output MOUNTPOINT,TYPE";
+    int exitCodeList = cloyster::runCommand(listCommand, output, false);
+
+    if (exitCodeList != 0) {
+        throw std::runtime_error("LVM ERROR: Failed to list mount points.");
+    }
+
+    bool isBootPartitionFound = false;
+
+    for (const auto& line : output) {
+        std::istringstream iss(line);
+        std::string mountPoint, type;
+        if (!(iss >> mountPoint >> type)) {
+            continue;
+        }
+
+        if (mountPoint == "/boot") {
+            isBootPartitionFound = true;
+            if (type == "lvm") {
+                m_hasBootPartition = true;
+            }
+        }
+    }
+
+    if (isBootPartitionFound && !m_hasBootPartition) {
+        LOG_INFO("LVM: /boot is mounted and not part of LVM.");
+    } else if (m_hasBootPartition) {
+        throw std::runtime_error("LVM ERROR: /boot is part of LVM. Check your "
+                                 "system configuration.");
```

Comment:
Should return a boolean and not throw. Also this condition may never be matched because if the system was able to boot with `/boot` in LVM this is not an issue. The point is that GRUB usually does not boot from LVM. So it ended up being irrelevant.